libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -963,6 +994,81 @@ def Delete(
             self.FormatLogMessage(
                 'Could not find disk: {0:s}, skipping'.format(disk_name)))
 
+  def AssignExternalIp(self, net_if: str, ip_addr: Optional[str]) -> None:
+    """Assigns an external IP to an instance's network interface.
+
+    The instance must not have an IP assigned to the network interface when
+    calling this method. If the IP address is specified, it must be one that
+    is available to the project.
+
+    Args:
+      net_if (str): The instance's network interface to which the IP address
+        must be assigned.
+      ip_addr (Optional[str]): The static IP address that exposes the network
+        interface. If None, the assigned IP address will be ephemeral.
+    """
+    body = {}
+    if ip_addr is not None:
+      body['natIP'] = ip_addr
+    instances_client = self.GceApi().instances()
+    request = instances_client.addAccessConfig(
+      project=self.project_id,
+      zone=self.zone,
+      instance=self.name,
+      networkInterface=net_if,
+      body=body
+    )
+    response = request.execute()
+    self.BlockOperation(response, self.zone)
+
+  def RemoveExternalIps(self) -> Dict[str, str]:
+    """
+    Removes any external IP of the instance, thus breaking
+    any ongoing connections.
+
+    Note that if the instance's IP address was static, that
+    the IP will still belong to the project.
+
+    Returns:
+      Dict[str, str]: A mapping from an instance's network
+        interfaces to the corresponding removed external IP.
+    """
+    external_ip_addresses = {}  # Result variable
+    # Iterate through instance's network interfaces, removing
+    # all access configurations (NAT)
+    instance_info = self.GetOperation()
+    for network_interface in instance_info['networkInterfaces']:
```

Comment:
Perhaps change `instance_info['networkInterfaces']` to `instance_info.get('networkInterfaces', [])`, so we make sure not to have unexpected `KeyErrors` here